### PR TITLE
adding additional schema for raw body data in uptime http check

### PR DESCRIPTION
**Pull request by chris-gilpatrick**
Adding an additional option to the uptime HTTP check to allow adding raw POST data through this Terraform module.

### CODE REVIEW

**Comment by chris-gilpatrick** [comment]
hi @tomasbasham could you approve the workflow please?

File: go.mod [hunk 1]
```diff
@@ -69,14 +69,14 @@ require (
...
-	golang.org/x/lint v0.0.0-20200302205851-738671d3881b // indirect
-	golang.org/x/mod v0.4.2 // indirect
+	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519 // indirect
+	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
+	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 // indirect
 	golang.org/x/net v0.0.0-20220121210141-e204ce36a2ba // indirect
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d // indirect
 	golang.org/x/sys v0.0.0-20220114195835-da31bd327af9 // indirect
 	golang.org/x/text v0.3.7 // indirect
-	golang.org/x/tools v0.1.2 // indirect
```
Comment (tomasbasham) [inline go.mod:79]: Could these be reverted back to the main branch as they're not really related to the schema change

File: internal/provider/resource_uptime_check.go [hunk 2]
```diff
@@ -183,6 +183,12 @@ func resourceStatusCakeUptimeCheck() *schema.Resource {
 								Type: schema.TypeString,
 							},
 						},
+						"request_raw": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+							Description:  "Raw POST data submitted with the request.",
```
Comment (tomasbasham) [inline internal/provider/resource_uptime_check.go:190]: To align this with the above schema element can this be changed to "Raw payload submitted with the request. Setting this updates the check to use the HTTP POST verb"

File: internal/provider/resource_uptime_check.go [hunk 3]
```diff
@@ -183,6 +183,12 @@ func resourceStatusCakeUptimeCheck() *schema.Resource {
 								Type: schema.TypeString,
 							},
 						},
+						"request_raw": &schema.Schema{
```
Comment (tomasbasham) [inline internal/provider/resource_uptime_check.go:186]: Can this be named `request_payload_raw`

Comment (tomasbasham) [reply internal/provider/resource_uptime_check.go:186]: Only one of `request_payload` and `request_payload_raw` may be included at any given time. This should be encapsulated in the schema.

**Review by tomasbasham — COMMENTED** [review]
Thanks for taking the time to fix this for us.

Just a couple comments in keeping with our internal naming conventions.

Also would you mind squashing the commits into a single commit in line with our [contribution guidelines](https://github.com/StatusCakeDev/terraform-provider-statuscake/blob/master/CONTRIBUTING.md#making-changes). Since this change only deals with a single issue a single commit containing the entire change will make our auditing easier.

File: internal/provider/resource_uptime_check.go (hunk 2 above)
Comment (chris-gilpatrick) [reply internal/provider/resource_uptime_check.go]: Description updated

File: go.mod (hunk 1 above)
Comment (chris-gilpatrick) [reply go.mod]: Changes reverted to match main branch values

**Comment by chris-gilpatrick** [comment]
Comments actioned and commits squashed into a single commit as per guidelines.

**Review by tomasbasham — APPROVED** [review]
Perfect. I'll look to getting this pushed into our testing pipeline and get it released

**Comment by chris-gilpatrick** [comment]
sorry - fixed an issue which was causing the tests to fail. i've also updated to ConflictsWith instead of ExactlyOne as both are optional

**Pull request merged**